pkg/cluster/executor/executor.go

```diff
@@ -36,3 +58,66 @@ type Executor interface {
 	// Transfer copies files from or to a target
 	Transfer(src string, dst string, download bool) error
 }
+
+// New create a new Executor
+func New(etype string, sudo bool, c SSHConfig) (Executor, error) {
+	// Used in integration testing, to check if native ssh client is really used when it need to be.
+	failpoint.Inject("assertNativeSSH", func() {
+		// XXX: We call system executor 'native' by mistake in commit f1142b1
+		// this should be fixed after we remove --native-ssh flag
+		if etype != SSHTypeSystem {
+			msg := fmt.Sprintf(
+				"native ssh client should be used in this case, os.Args: %s, %s = %s",
+				os.Args, localdata.EnvNameNativeSSHClient, os.Getenv(localdata.EnvNameNativeSSHClient),
+			)
+			panic(msg)
+		}
+	})
+
+	// set default values
+	if c.Port <= 0 {
+		c.Port = 22
+	}
+
+	if c.Timeout == 0 {
+		c.Timeout = time.Second * 5 // default timeout is 5 sec
+	}
+
+	switch etype {
+	case SSHTypeBuiltin:
+		e := &EasySSHExecutor{
+			Locale: "C",
+			Sudo:   sudo,
+		}
+		e.initialize(c)
+		return e, nil
+	case SSHTypeSystem:
+		e := &NativeSSHExecutor{
+			Config: &c,
+			Locale: "C",
+			Sudo:   sudo,
+		}
+		if c.Password != "" || (c.KeyFile != "" && c.Passphrase != "") {
+			_, _, e.ConnectionTestResult = e.Execute(connectionTestCommand, false, executeDefaultTimeout)
+		}
+		return e, nil
+	case SSHTypeNone:
+		local := os.Getenv(localdata.EnvNameLocalHost)
```

Comment:
I don't think we should use an envvar to figure out the local IP. It's better to use
```go
ifaces, err := net.Interfaces()
// handle err
for _, i := range ifaces {
    addrs, err := i.Addrs()
    // handle err
    for _, addr := range addrs {
        var ip net.IP
        switch v := addr.(type) {
        case *net.IPNet:
                ip = v.IP
        case *net.IPAddr:
                ip = v.IP
        }
        // process IP address
    }
}
```
Check whether in the local IP list more naturally.

Reply:
Can this handle elastic IP address?

Reply:
I don't think we need to handle the elastic IP for local mode because there will not be inter-connect with other nodes.